crypto/ec/ecp_nistz256.c

```diff
@@ -358,16 +358,47 @@ static void ecp_nistz256_point_add(P256_POINT *r,
     ecp_nistz256_sub(H, U2, U1);                /* H = U2 - U1 */
 
     /*
-     * This should not happen during sign/ecdh, so no constant time violation
+     * The formulae are incorrect if the points are equal so we check for
+     * this and do doubling if this happens.
+     *
+     * Points here are in Jacobian projective coordinates (Xi, Yi, Zi)
+     * that are bound to the affine coordinates (xi, yi) by the following
+     * equations:
+     *     - xi = Xi / (Zi)^2
+     *     - y1 = Yi / (Zi)^3
+     *
+     * For the sake of optimization, the algorithm operates over
+     * intermediate variables U1, U2 and S1, S2 that are derived from
+     * the projective coordinates:
+     *     - U1 = X1 * (Z2)^2 ; U2 = X2 * (Z1)^2
+     *     - S1 = Y1 * (Z2)^3 ; U2 = Y2 * (Z1)^3
```

Comment:
Is U2 = Y2 * (Z1)^3 supposed to be S2 = Y2 * (Z1)^3

Reply:
Yes, indeed, this line should say `S1 = Y1 * (Z2)^3 ; S2 = Y2 * (Z1)^3`

Reply:
I will adjust this comment! Thanks for slotting it @zorrorffm

I also saw your PR with the arm8 changes, great work, thanks! 

I plan to take this PR out of WIP today as at the last face-to-face meeting it was decided to close this PR by merging the fixes for the platforms that are currently addressed and open issues for the remaining fixes so the community could help.

Would it be fine for you @zorrorffm if I cherry-picked your commit from #10729 as I wrap this PR up for approval?

Reply:
The comment was fixed.